### ISSUE TITLE
Semantic cast drops metadata for temporary eager constants

### ISSUE THREAD

**Issue by shinaoka** [tensor4all/tenferro-rs#1700]
## Summary

A mixed-dtype eager operation can fail during backward when one promoted input is a non-leaf semantic trace that depends on a temporary untracked eager constant. The eager primal succeeds, but deferred AD compilation later reports missing metadata for that temporary input.

## Minimal reproducer

Using tenferro revision `8a8196a95363158f147b1feff2bc3b2d4bc4d267`:

```rust
use num_complex::Complex64;
use tenferro_ad::{EagerRuntime, EagerTensor};
use tenferro_cpu::CpuBackend;
use tenferro_runtime::Tensor;

fn main() -> Result<(), Box<dyn std::error::Error>> {
    let ctx = EagerRuntime::with_cpu_backend(CpuBackend::new())?;
    let x = EagerTensor::requires_grad_in(
        Tensor::from_vec_col_major(vec![], vec![-4.0_f64])?,
        ctx.clone(),
    )?;
    let y = {
        let magnitude_input = x.neg()?;
        let magnitude = {
            let exponent = EagerTensor::from_tensor_in(
                Tensor::from_vec_col_major(vec![], vec![0.5_f64])?,
                ctx.clone(),
            )?;
            magnitude_input.pow(&exponent)?
        };
        let factor = EagerTensor::from_tensor_in(
            Tensor::from_vec_col_major(vec![], vec![Complex64::new(0.0, 1.0)])?,
            ctx,
        )?;
        magnitude.mul(&factor)?
    };
    y.backward()?;
    assert!(x.grad()?.is_some());
    Ok(())
}
```

## Actual result

```text
metadata (Compile): runtime state failure: metadata registration:
missing input metadata for Input(User { id: 4 })
```

## Expected result

`backward()` succeeds and returns the real projection of the complex cotangent for `x`.

## Suspected cause

The failure requires the combination of:

1. an untracked temporary input in an active eager semantic trace (`pow` exponent), and
2. a later mixed-dtype operation that semantically casts the non-leaf result (`F64 * C64`).

A direct tracked `F64 *` temporary `C64` succeeds, as does tracked `F64.pow()` with a temporary `F64` exponent. This suggests that semantic `Convert` insertion for the mixed operation preserves the traced graph but does not preserve all metadata registrations/guards inherited from the cast input's non-leaf trace.

## Downstream impact

This blocks tensor4all-rs #623 on the final synchronized tenferro revision. Tensor4all's tracked principal-branch implementation for negative real `powf` naturally builds this graph, and no downstream materialization or retained-temporary workaround is appropriate because it would either detach AD or hide the eager trace-lifetime defect.
